hal/network/ncp_client/sara/sara_ncp_client.cpp

```diff
@@ -966,6 +967,126 @@ int SaraNcpClient::checkNetConfForImsi() {
     return SYSTEM_ERROR_TIMEOUT;
 }
 
+int SaraNcpClient::selectNetworkProf(ModemState& state) {
+    bool reset = false;
```

Comment:
Nit: consider moving this and `disableLowPowerModes` a bit further down before the loop/inside the loop where it becomes relevant?